relay-server/src/service.rs

```diff
@@ -464,15 +474,33 @@ pub fn create_redis_pools(configs: RedisPoolConfigs) -> Result<RedisPools, Redis
     }
 }
 
+#[cfg(feature = "processing")]
+async fn create_async_pool(config: &RedisConfigRef<'_>) -> Result<AsyncRedisPool, RedisError> {
+    match config {
+        RedisConfigRef::Cluster {
+            cluster_nodes,
+            options,
+        } => AsyncRedisPool::cluster(cluster_nodes.iter().map(|s| s.as_str()), options).await,
+        RedisConfigRef::Single { server, options } => {
+            AsyncRedisPool::single(server.as_str(), options).await
+        }
+        RedisConfigRef::MultiWrite { configs } => {
+            // Based on the assumption that the first config is the
+            // primary config for MultiWrite configurations
+            let primary_config = configs.iter().next().ok_or(RedisError::Configuration)?;
+            Box::pin(create_async_pool(primary_config)).await
```

Comment:
Not sure why you have the `Box::pin` here, if your objective is to pin a future, you should pin it using `std::pin::pin!`

Reply:
I need to have a boxed future for the recursion to work. Using `std::pin::pin!` doesn't work and produces this: `recursion in an async fn requires boxing`